x-pack/plugins/security_solution/public/common/utils/global_query_string/index.ts

```diff
@@ -41,10 +41,14 @@ export const useInitializeUrlParam = <State>(
   onInitialize: (state: State | null) => void
 ) => {
   const dispatch = useDispatch();
-  const { search } = useLocation();
 
   useEffect(() => {
-    const initialValue = getParamFromQueryString(getQueryStringFromLocation(search), urlParamKey);
+    // window.location.search provides the most updated representation of the url search.
+    // It also guarantees that we don't overwrite URL param managed outside react-router.
+    const initialValue = getParamFromQueryString(
+      getQueryStringFromLocation(window.location.search),
```

Comment:
I had to use `window.location.search` to fix a bug. After everything is migrated to the new API, we will reconsider this initialization approach.